WooCommerce/WooCommerceTests/ViewRelated/Search/Customer/CustomerSearchUICommandTests.swift

```diff
@@ -61,4 +73,32 @@ final class CustomerSearchUICommandTests: XCTestCase {
         // Then
         XCTAssert(analyticsProvider.receivedEvents.contains("order_creation_customer_search"))
     }
+
+    func test_synchronizeModels_when_better_customer_selection_is_enabled_and_keyword_is_empty_then_calls_synchronizeAllLightCustomersDataAction() {
+        // Given
+        let stores = MockStoresManager(sessionManager: .testingInstance)
+        // Given
+        let command = CustomerSearchUICommand(siteID: sampleSiteID,
+                                              stores: stores,
+                                              featureFlagService: MockFeatureFlagService(betterCustomerSelectionInOrder: true)) { _ in }
+
+        var invocationCount = 0
+        stores.whenReceivingAction(ofType: CustomerAction.self) { action in
+            guard case let .synchronizeLightCustomersData(_, _, _, onCompletion) = action else {
+                return XCTFail("Unexpected action: \(action)")
+            }
+            invocationCount += 1
+            onCompletion(.success(()))
+        }
+
+        // When
+        waitFor { promise in
+            command.synchronizeModels(siteID: self.sampleSiteID, keyword: "", pageNumber: 1, pageSize: 10) { _ in
+                promise(())
+            }
+        }
```

Comment:
Just a note here, since I'm not sure why is happening: I ran this test a few times without the `waitFor` block (but still calling `command.synchronizerModels()`) and keeps passing correctly. I would expect that fails because of the async work that must happen when we dispatch the action, but that's not the case 🤔

Reply:
That's a good question, perhaps it stays in the same thread as the `synchronizeLightCustomersData` returns immediately in this test? In this case I would leave the `waitFor` in case it's changed in the future.